packages/flame/test/components/component_test.dart

```diff
@@ -199,6 +205,111 @@ void main() {
         equals([Vector2(300, 200), Vector2(400, 500)]),
       );
     });
+    testWithFlameGame(
+      'length of descendants must be 0 '
+      'when it is called before the add method',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+
+        expect(game.descendants().length, 0);
+
+        await game.ensureAdd(component);
+
+        expect(game.descendants().length, 3);
```

Comment:
You mean that the test above is similar to this one, therefore this test is meaningless?
If i change the name to?
`should return the correct length of the descendants after the onLoaded method finishes`

Reply:
I think this test is great!

I think we can leave it as it is. Great job @Dan-Crane !

I marked this as nit because from the test name (which I like) I was not expecting to see the last `expect` but I do find value on the last `expect`, since it's verifying that there are no aide effects on calling it again after being called before loaded.

Reply:
I am not offended if i wrote bad code/comment, speak up) it will help me improve)

yes, this is a test that everything went well